reframe/frontend/statistics.py

```diff
@@ -315,42 +313,63 @@ def print_failure_stats(self, printer):
                 printer.info(line)
 
     def performance_report(self):
-        # FIXME: Adapt this function to use the JSON report
-
-        line_width = 78
-        report_start = line_width * '='
-        report_title = 'PERFORMANCE REPORT'
-        report_end = line_width * '-'
-        report_body = []
-        previous_name = ''
-        previous_part = ''
-        for t in self.tasks():
-            if t.check.perfvalues.keys():
-                if t.check.unique_name != previous_name:
-                    report_body.append(line_width * '-')
-                    report_body.append(t.check.display_name)
-                    previous_name = t.check.unique_name
-
-                if t.check.current_partition.fullname != previous_part:
-                    report_body.append(
-                        f'- {t.check.current_partition.fullname}')
-                    previous_part = t.check.current_partition.fullname
-
-                report_body.append(f'   - {t.check.current_environ}')
-                report_body.append(f'      * num_tasks: {t.check.num_tasks}')
-
-            for key, ref in t.check.perfvalues.items():
-                var = key.split(':')[-1]
-                val = ref[0]
-                try:
-                    unit = ref[4]
-                except IndexError:
-                    unit = '(no unit specified)'
-
-                report_body.append(f'      * {var}: {val} {unit}')
-
-        if report_body:
-            return '\n'.join([report_start, report_title, *report_body,
-                              report_end])
-
-        return ''
+        width = shutil.get_terminal_size()[0]
+        lines = ['', width*'=', 'PERFORMANCE REPORT', width*'-']
+
+        # Collect all the records from performance tests
+        perf_records = {}
+        for run in self.json():
+            for tc in run['testcases']:
+                if tc['perfvars']:
+                    key = tc['unique_name']
```

Comment:
I think this is why the performance report contains only one env-system per test
```suggestion
                    key = (tc['unique_name'], tc['system'], tc['environment'])
```

Reply:
oh, sorry, my bad. Didn't realize you had already fixed the bug.